packages/core/feature-flags/core-feature-flags-browser-internal/src/feature_flags_service.ts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { CoreContext } from '@kbn/core-base-browser-internal';
+import type { InternalInjectedMetadataSetup } from '@kbn/core-injected-metadata-browser-internal';
+import type { Logger } from '@kbn/logging';
+import type {
+  EvaluationContext,
+  FeatureFlagsSetup,
+  FeatureFlagsStart,
+} from '@kbn/core-feature-flags-browser';
+import { apm } from '@elastic/apm-rum';
+import { type Client, ClientProviderEvents, OpenFeature } from '@openfeature/web-sdk';
+import deepMerge from 'deepmerge';
+import { filter, map, startWith, Subject } from 'rxjs';
+
+/**
+ * setup method dependencies
+ * @private
+ */
+export interface FeatureFlagsSetupDeps {
+  /**
+   * Used to read the flag overrides set up in the configuration file.
+   */
+  injectedMetadata: InternalInjectedMetadataSetup;
+}
+
+export class FeatureFlagsService {
+  private readonly featureFlagsClient: Client;
+  private readonly logger: Logger;
+  private isProviderReadyPromise?: Promise<void>;
+  private context: EvaluationContext = { kind: 'multi' };
+  private overrides: Record<string, unknown> = {};
+
+  constructor(core: CoreContext) {
+    this.logger = core.logger.get('feature-flags-service');
+    this.featureFlagsClient = OpenFeature.getClient();
+    OpenFeature.setLogger(this.logger.get('open-feature'));
```

Comment:
Core packages are loaded as singletons on the front end right? Just wondering if `@openfeature/web-sdk` being loaded more than once will lead to issues.

Reply:
AFAIK, `core` is a bundle on its own that's only loaded once.

Reply:
Got it, would it be an issue if some other non-core code imports `@openfeature/web-sdk`?

Reply:
It might be... but I guess it won't work for them, because they'll get a new client in their bundle and will fallback to our core APIs (or that I hope 😅)